core/resharding.go

```diff
@@ -1,259 +0,0 @@
-package core
-
-import (
-	"encoding/hex"
-	"errors"
-	"math/big"
-	"math/rand"
-	"sort"
-
-	"github.com/ethereum/go-ethereum/common"
-	"github.com/harmony-one/bls/ffi/go/bls"
-	common2 "github.com/harmony-one/harmony/internal/common"
-	shardingconfig "github.com/harmony-one/harmony/internal/configs/sharding"
-	"github.com/harmony-one/harmony/internal/ctxerror"
-	"github.com/harmony-one/harmony/internal/utils"
-	"github.com/harmony-one/harmony/shard"
-)
-
-const (
-	// GenesisEpoch is the number of the genesis epoch.
-	GenesisEpoch = 0
-	// CuckooRate is the percentage of nodes getting reshuffled in the second step of cuckoo resharding.
-	CuckooRate = 0.1
-)
-
-// ShardingState is data structure hold the sharding state
-type ShardingState struct {
-	epoch      uint64 // current epoch
-	rnd        uint64 // random seed for resharding
-	numShards  int    // TODO ek – equal to len(shardState); remove this
-	shardState shard.State
-}
-
-// sortedCommitteeBySize will sort shards by size
-// Suppose there are N shards, the first N/2 larger shards are called active committees
-// the rest N/2 smaller committees are called inactive committees
-// actually they are all just normal shards
-// TODO: sort the committee weighted by total staking instead of shard size
-func (ss *ShardingState) sortCommitteeBySize() {
-	sort.Slice(ss.shardState, func(i, j int) bool {
-		return len(ss.shardState[i].NodeList) > len(ss.shardState[j].NodeList)
-	})
-}
-
-// assignNewNodes add new nodes into the N/2 active committees evenly
-func (ss *ShardingState) assignNewNodes(newNodeList []shard.NodeID) {
-	ss.sortCommitteeBySize()
-	numActiveShards := ss.numShards / 2
-	Shuffle(newNodeList)
-	for i, nid := range newNodeList {
-		id := 0
-		if numActiveShards > 0 {
-			id = i % numActiveShards
-		}
-		if id < len(ss.shardState) {
-			ss.shardState[id].NodeList = append(ss.shardState[id].NodeList, nid)
-		} else {
-			utils.Logger().Error().Int("id", id).Int("shardState Count", len(ss.shardState)).Msg("assignNewNodes index out of range")
-		}
-	}
-}
-
-// cuckooResharding uses cuckoo rule to reshard X% of active committee(shards) into inactive committee(shards)
-func (ss *ShardingState) cuckooResharding(percent float64) {
```

Comment:
Why remove all the resharding code? Some will be needed when we do the actual resharding.

Reply:
Because it was confusing, not used and now new setup of resharding is ready. Can always go back in git history to pull this code. Having less code in place makes it easier to understand what is happening in big picture because don't need to worry if used being used or not. We can have a new package under shard, (like committee) that can provide own place; makes `core` smaller